chain/jsonrpc/src/lib.rs

```diff
@@ -283,16 +287,68 @@ impl JsonRpcHandler {
         .map_err(|_| timeout_err())?
     }
 
+    async fn send_tx(
+        &self,
+        tx: SignedTransaction,
+        check_only: bool,
+    ) -> Result<NetworkClientResponses, RpcError> {
+        Ok(self
+            .client_addr
+            .send(NetworkClientMessages::Transaction {
+                transaction: tx,
+                is_forwarded: false,
+                check_only,
+            })
+            .map_err(|err| RpcError::server_error(Some(ServerError::from(err))))
+            .await?)
+    }
+
+    async fn send_tx_sync(&self, params: Option<Value>) -> Result<Value, RpcError> {
+        self.send_or_check_tx(params, false).await
+    }
+
+    async fn check_tx(&self, params: Option<Value>) -> Result<Value, RpcError> {
+        self.send_or_check_tx(params, true).await
+    }
+
+    async fn send_or_check_tx(
+        &self,
+        params: Option<Value>,
+        check_only: bool,
+    ) -> Result<Value, RpcError> {
+        let tx = parse_tx(params)?;
+        let tx_hash = (&tx.get_hash()).to_base();
+        match self.send_tx(tx, check_only).await? {
+            NetworkClientResponses::ValidTx => {
+                if check_only {
+                    Ok(Value::Null)
+                } else {
+                    Ok(Value::String(tx_hash))
+                }
+            }
+            NetworkClientResponses::RequestRouted => {
+                if check_only {
+                    Ok(Value::String("Node doesn't track this shard. Cannot determine whether the transaction is valid".to_string()))
+                } else {
+                    Ok(Value::String("Transaction is routed".to_string()))
+                }
+            }
+            NetworkClientResponses::InvalidTx(err) => {
+                Err(RpcError::server_error(Some(ServerError::TxExecutionError(err.into()))))
+            }
+            NetworkClientResponses::NoResponse => {
+                // this is only possible if something went wrong with the node internally.
+                Err(RpcError::server_error(Some(ServerError::InternalError)))
+            }
+            _ => unreachable!(),
```

Comment:
You suggested previously that we should not crash the node if some RPC fails. I guess it is a good idea to just respond with InternalError here.